warp10/src/main/java/io/warp10/standalone/StandaloneAcceleratedStoreClient.java

```diff
@@ -0,0 +1,298 @@
+          this.cache.store(encoder);
+          datapoints.addAndGet(decoder.getCount());
+        }
+      }
+
+      exec.shutdown();
+      
+      System.out.println("Preloaded accelerator with " + datapoints + " datapoints from " + this.cache.getGTSCount() + " Geo Time Series in " + ((System.nanoTime() - nanos) / 1000000.0D) + " ms.");
+      LOG.info("Preloaded accelerator with " + datapoints + " datapoints from " + this.cache.getGTSCount() + " Geo Time Series in " + ((System.nanoTime() - nanos) / 1000000.0D) + " ms.");
+    } catch (IOException ioe) {
+      throw new RuntimeException("Error populating cache.", ioe);
+    }
+  }
+  
+  @Override
+  public void addPlasmaHandler(StandalonePlasmaHandlerInterface handler) {
+    this.persistent.addPlasmaHandler(handler);
+  }
+  
+  @Override
+  public long delete(WriteToken token, Metadata metadata, long start, long end) throws IOException {
+    cache.delete(token, metadata, start, end);
+    persistent.delete(token, metadata, start, end);
+    return 0;
+  }
+  
+  @Override
+  public GTSDecoderIterator fetch(ReadToken token, List<Metadata> metadatas, long now, long then, long count, long skip, double sample, boolean writeTimestamp, int preBoundary, int postBoundary) throws IOException {
+    //
+    // If the fetch has both a time range that is larger than the cache range, we will only use
+    // the persistent backend to ensure a correct fetch. Same goes with boundaries which could extend outside the
+    // cache.
+    //
+    // Note that this is a heuristic which could still lead to missing datapoints as data with timestamps within
+    // the current cache time range could very well have been written to the persistent store and not preloaded
+    // at cache startup if they were not in an active chunk of the cache.
+    //
+    
+    long cacheend = InMemoryChunkSet.chunkEnd(now, this.cache.getChunkSpan());
+    long cachestart = cacheend - this.cache.getChunkCount() * this.cache.getChunkSpan() + 1;
+
+    //
+    // If fetching a single value from Long.MAX_VALUE with an ephemeral cache, always use the cache
+    // unless ACCEL.NOCACHE was called.
+    //
+    if (this.ephemeral && 1 == count && Long.MAX_VALUE == now && !nocache.get()) {
+      accelerated.set(Boolean.TRUE);
+      return this.cache.fetch(token, metadatas, now, then, count, skip, sample, writeTimestamp, preBoundary, postBoundary);      
+    }
+    
+    // Use the persistent store unless ACCEL.NOPERSIST was called 
+    if (((now > cacheend || then < cachestart) || preBoundary > 0 || postBoundary > 0 || nocache.get()) && !nopersist.get()) {
+      accelerated.set(Boolean.FALSE);
+      return this.persistent.fetch(token, metadatas, now, then, count, skip, sample, writeTimestamp, preBoundary, postBoundary);
+    }
+    
+    // Last resort, use the cache
+    accelerated.set(Boolean.TRUE);
+    return this.cache.fetch(token, metadatas, now, then, count, skip, sample, writeTimestamp, preBoundary, postBoundary);
```

Comment:
Should the result be deduped to have consistent result between cache and persist?
Or modify the cache so it does not store duplicate ticks.

Also the result will not be sorted using the cache but will be using persist. I don't think this is as problematic as the duplicate ticks is.

Reply:
The semantics are those of the in-memory store, duplication is not handled at the store level, for the rare cases where you want to avoid duplicates datapoints, simply call DEDUP after FETCH if it was against the in-memory store.